AutomatedTesting/Gem/PythonTests/Physics/tests/EntityComponentTests/PhysX_Collider_Component_CRUD.py

```diff
@@ -0,0 +1,167 @@
+"""
+Copyright (c) Contributors to the Open 3D Engine Project.
+For complete copyright and license terms please see the LICENSE at the root of this distribution.
+
+SPDX-License-Identifier: Apache-2.0 OR MIT
+"""
+
+def PhysX_Collider_Component_CRUD():
+    """
+    Summary:
+    Creating a PhysX Collider and setting values to all properties that are supported via the Editor Context Bus.
+
+    Expected Behavior:
+    Validates that Component CRUD can be performed without crashing.
+
+    PhysX Collider Components:
+     - PhysX Collider
+
+    Test Steps:
+     1) Add an Entity to manipulate
+     2) Add a PhysX Collider Component to Entity
+     3) Set Box Shape and Child Properties
+     4) Set Capsule Shape and Child Properties
+     5) Set Cylinder Shape and Child Properties
+     6) Set Sphere Shape and Child Properties
+     7) Set General Properties
+     8) Set PhyicsAsset Shape and Child Properties
+     9) Delete Component
+
+    :return: None
+    """
+
+    # Helper file Imports
+    import os
+
+    from editor_python_test_tools.editor_component.physx_collider import PhysxCollider
+    from editor_python_test_tools.editor_entity_utils import EditorEntity
+    from editor_python_test_tools.utils import Tracer, TestHelper
+
+    from consts.general import Strings
+
+    # 0) Pre-conditions
+    physx_mesh = os.path.join("objects", "_primitives", "_box_1x1.pxmesh")
+    physx_material = os.path.join("physx", "glass.physxmaterial")
+
+    TestHelper.init_idle()
+    TestHelper.open_level("", "Base")
+
+    with Tracer() as section_tracer:
+    # 1 ) Add an Entity to manipulate
+        phsyx_collider_entity = EditorEntity.create_editor_entity("TestEntity")
+
+    # 2) Add a PhysX Collider Component to Entity
+        physx_collider = PhysxCollider(phsyx_collider_entity)
+
+    # 3) Set Box Shape and Child Properties
+        physx_collider.set_box_shape()
+
+        physx_collider.set_box_dimensions(256.0, 256.0, 256.0)
+        physx_collider.set_box_dimensions(2.5, 2.5, 2.5)
+        physx_collider.set_box_dimensions(0.0, 0.0, 0.0)
+        physx_collider.set_box_dimensions(-1.0, -10, -256.0)
```

Comment:
question: Having all these operations in one large test does seem a little strange to me (unless we are treating each API call as effectively a 'test'). This test now has many reasons to fail which can make debugging it difficult. Could you include some example failures say if `set_box_dimensions` was to fail for various reasons (e.g. out of range or an invalid value of some kind etc...). It would be good to get an idea of what failures look like and how easy they'll be to identify and fix.

Reply:
Using asserts, you get a python trace.

`E       EXCEPTION raised:
E         Traceback (most recent call last):
E           File "e:\gws\o3de\automatedtesting\gem\pythontests\editorpythontesttools\editor_python_test_tools\utils.py", line 328, in start_test
E             test_function()
E           File "E:/gws/o3de/AutomatedTesting/Gem/PythonTests/Physics/tests/EntityComponentTests/PhysX_Collider_Component_CRUD.py", line 126, in PhysX_Collider_Component_CRUD
E             physx_collider.set_tag(Strings.NUMBER)
E           File "e:\gws\o3de\automatedtesting\gem\pythontests\editorpythontesttools\editor_python_test_tools\editor_component\physx_collider.py", line 122, in set_tag
E             self.component.set_component_property_value(self.Path.TAG, tag)
E           File "e:\gws\o3de\automatedtesting\gem\pythontests\editorpythontesttools\editor_python_test_tools\editor_entity_utils.py", line 264, in set_component_property_value
E             assert (
E         AssertionError: Failure: Could not set value to 'PhysX Collider' : 'Collider Configuration|Tag'
E       Test result:  FAILURE`

If we're logging tracer errors for validating failing gracefully with negative conditions, we'd have to develop that pattern.